pkgs/development/libraries/libgnt/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, lib, fetchurl, meson, ninja, pkg-config, python2
+, gtk-doc, docbook-xsl-nons
+, glib, ncurses, libxml2
+, buildDocs ? true
+}:
+stdenv.mkDerivation rec {
+  pname = "libgnt";
+  version = "2.14.1";
+
+  outputs = [ "out" "dev" ] ++ lib.optional buildDocs "devdoc";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/pidgin/${pname}-${version}.tar.xz";
+    sha256 = "1n2bxg0ignn53c08cp69pj4sdg53kwlqn23rincyjmpr327fdhsy";
+  };
+
+  nativeBuildInputs = [ meson ninja pkg-config ]
+    ++ lib.optionals buildDocs [ gtk-doc docbook-xsl-nons ];
+
+  # python2 is optional but Pidgin does depend on it (and python3 won't be accepted in 2.x)
+  buildInputs = [ glib ncurses libxml2 python2 ];
```

Comment:
How does that work when you are using python3 in Pidgin?

Reply:
Thank you for spotting this. I just semi-blindly cherry-picket this.  
As per https://keep.imfreedom.org/libgnt/libgnt it should be Python 3.x . But when I switched to 3.x I'm getting:

```
Run-time dependency python2 found: NO (tried pkgconfig and cmake)
```

Maybe it works because scripts are still compatible between 2.x and 3.x or because it actually not used until particular plugins installed.  
The only reference to finch and Python I found is [here](https://keep.imfreedom.org/pidgin/pidgin/file/7f87a83083b0bb4f4e87d0b3f26ad6d86d7a3108/ChangeLog#l394). Their web interface is unusable for tracing back to commit in order to figure out what exactly was done there.

P.S. It seems that [Pidgin depends on Python 3.x directly for code generation](https://keep.imfreedom.org/pidgin/pidgin/file/7f87a83083b0bb4f4e87d0b3f26ad6d86d7a3108/configure.ac#l1435) and for [DBus scripts](https://keep.imfreedom.org/pidgin/pidgin/file/7f87a83083b0bb4f4e87d0b3f26ad6d86d7a3108/ChangeLog#l59) in run-time.

Reply:
This libgnt release is python2 only, apparently
```meson
python_dep = dependency('python2', required : false)
```

Reply:
Though their hg repo already has some version with python3 (only) support: https://keep.imfreedom.org/libgnt/libgnt/file/default/meson.build#l208

Reply:
Would replacing it for `python3` in `meson.build` fix it?

Reply:
It would build with python 3.7, but our default is 3.8 (and 3.9 in `staging-next` branch) and there it fails to link.

Reply:
Maybe it will be good without any python as well, e.g. Debian doesn't specify any: https://packages.debian.org/source/bullseye/libgnt

Reply:
After removing completely and building (`nix-build -A pidgin`) `finch` still works (`./result/bin/finch`) and shows accounts.